src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs

```diff
@@ -0,0 +1,286 @@
+            if (rc == -1)
+            {
+                throw new PlatformNotSupportedException(string.Format(SR.Cryptography_CurveNotSupported, oid));
+            }
+            return key;
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByExplicitParameters")]
+        internal static extern SafeEcKeyHandle EcKeyCreateByExplicitParameters(
+            ECCurve.ECCurveType curveType,
+            byte[] qx, int qxLength,
+            byte[] qy, int qyLength,
+            byte[] d, int dLength,
+            byte[] p, int pLength,
+            byte[] a, int aLength,
+            byte[] b, int bLength,
+            byte[] gx, int gxLength,
+            byte[] gy, int gyLength,
+            byte[] order, int nLength,
+            byte[] cofactor, int cofactorLength,
+            byte[] seed, int seedLength);
+
+        internal static SafeEcKeyHandle EcKeyCreateByExplicitCurve(ECCurve curve)
+        {
+            byte[] p;
+            if (curve.IsPrime)
+            {
+                p = curve.Prime;
+            }
+            else if (curve.IsCharacteristic2)
+            {
+                p = curve.Polynomial;
+            }
+            else
+            {
+                throw new PlatformNotSupportedException(string.Format(SR.Cryptography_CurveNotSupported, curve.CurveType.ToString()));
+            }
+
+            SafeEcKeyHandle key = Interop.Crypto.EcKeyCreateByExplicitParameters(
+                curve.CurveType,
+                null, 0,
+                null, 0,
+                null, 0,
+                p, p.Length,
+                curve.A, curve.A.Length,
+                curve.B, curve.B.Length,
+                curve.G.X, curve.G.X.Length,
+                curve.G.Y, curve.G.Y.Length,
+                curve.Order, curve.Order.Length,
+                curve.Cofactor, curve.Cofactor.Length,
+                curve.Seed, curve.Seed == null ? 0 : curve.Seed.Length);
+
+            if (key == null || key.IsInvalid)
+            {
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+            }
+
+            return key;
+        }
+
+
+        [DllImport(Libraries.CryptoNative)]
+        private static extern bool CryptoNative_GetECKeyParameters(
+            SafeEcKeyHandle key, 
+            bool includePrivate,
+            out SafeBignumHandle qx_bn, out int x_cb,
+            out SafeBignumHandle qy_bn, out int y_cb,
+            out IntPtr d_bn_not_owned, out int d_cb);
+
+        internal static ECParameters GetECKeyParameters(
+            SafeEcKeyHandle key,
+            bool includePrivate)
+        {
+            SafeBignumHandle qx_bn, qy_bn;
+            IntPtr d_bn_not_owned;
+            int qx_cb, qy_cb, d_cb;
+            ECParameters parameters = new ECParameters();
+
+            bool refAdded = false;
+            key.DangerousAddRef(ref refAdded); // Protect access to d_bn_not_owned
```

Comment:
According to the pattern, this should be done inside the try block as the first thing.  In practice, it's not going to matter here as the purpose is to protect against thread aborts, which don't exist in .NET Core.  Still, for consistency and because there's no downside, I'd suggest moving this to be inside the try.